src/orange/surf/RecursiveSimplifier.hh

```diff
@@ -0,0 +1,98 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/surf/RecursiveSimplifier.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <utility>
+#include <variant>
+
+#include "SurfaceSimplifier.hh"
+#include "VariantSurface.hh"
+#include "detail/AllSurfaces.hh"
+#include "detail/RecursiveSimplifierImpl.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Recursively simplify and call the given function on the final surface.
+ *
+ * Example:
+ * \code
+  RecursiveSimplifier print_simplified([](Sense s, auto&& surf) {
+      cout << to_char(s) << surf << endl;
+  }, 1e-10);
+  print_simplified(Sense::inside, Plane{{1,0,0}, 4});
```

Comment:
Delete this line

Reply:
This shows the templated `operator()` usage; the one below shows the variant usage. I'll add documentation to point out the difference.